crates/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1117,10 +1167,180 @@ impl Context {
                 };
                 Ok(vec![AcirValue::Var(self.acir_context.add_constant(len), AcirType::field())])
             }
+            Intrinsic::SlicePushBack => {
+                let slice_length = self.convert_value(arguments[0], dfg).into_var()?;
+                let slice = self.convert_value(arguments[1], dfg);
+                // TODO(#2461): make sure that we have handled nested struct inputs
+                let element = self.convert_value(arguments[2], dfg);
+
+                let one = self.acir_context.add_constant(FieldElement::one());
+                let new_slice_length = self.acir_context.add_var(slice_length, one)?;
+
+                let mut new_slice = Vector::new();
+                self.slice_intrinsic_input(&mut new_slice, slice)?;
+                new_slice.push_back(element);
+
+                Ok(vec![
+                    AcirValue::Var(new_slice_length, AcirType::field()),
+                    AcirValue::Array(new_slice),
+                ])
+            }
+            Intrinsic::SlicePushFront => {
+                let slice_length = self.convert_value(arguments[0], dfg).into_var()?;
+                let slice = self.convert_value(arguments[1], dfg);
+                // TODO(#2461): make sure that we have handled nested struct inputs
```

Comment:
Orthogonal to this PR; 2461 is marked as E-low, will there be a follow-up PR to fix this?

Reply:
Yes there will be, after finishing this PR though I chose to focus on fixing bugs we had rather than adding new features as this PR gets slices up-to-date with the current array functionality. I felt that adding support for dynamic indices with non-homogenous arrays/slices could be its own PR.